api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2Client.java

```diff
@@ -209,12 +211,19 @@ private void renewTokens() {
             // if that fails, of if tokens weren't available, try fetching brand-new tokens
             .exceptionally(error -> fetchNewTokens());
     currentTokensStage
-        .whenComplete((tokens, error) -> log(error))
+        .whenComplete((tokens, error) -> maybeHandleError(error))
         .whenComplete((tokens, error) -> maybeScheduleTokensRenewal(tokens));
   }
 
-  private void log(Throwable error) {
+  private void maybeHandleError(Throwable error) {
     if (error != null) {
+      if (hasCauseMatching(error, t -> t instanceof TimeoutException)) {
+        LOGGER.debug("Closing due to {}", error.toString());
```

Comment:
Include `error` in the log message?

Reply:
meh - it'll already show you the exception class and message - guess that's enough